pkg/command-handler/domain/ledger/usecase/create_transaction_test.go

```diff
@@ -0,0 +1,195 @@
+
+		err := newFakeUseCase(nil).CreateTransaction(context.Background(), uuid.New(), entries)
+
+		assert.Nil(t, err)
+	})
+
+	t.Run("Successfully creates a transaction with expected version", func(t *testing.T) {
+		useCase := newFakeUseCase(nil)
+
+		accountID1 := uuid.New()
+		accountID2 := uuid.New()
+		e1 := entities.NewEntry(uuid.New(), entities.DebitOperation, accountID1, entities.NewAccountVersion, 123)
+		e2 := entities.NewEntry(uuid.New(), entities.CreditOperation, accountID2, entities.NewAccountVersion, 123)
+		entries := []entities.Entry{*e1, *e2}
+
+		err := useCase.CreateTransaction(context.Background(), uuid.New(), entries)
+
+		assert.Nil(t, err)
+
+		e1 = entities.NewEntry(uuid.New(), entities.DebitOperation, accountID1, 2, 123)
+		e2 = entities.NewEntry(uuid.New(), entities.CreditOperation, accountID2, 3, 123)
+		entries = []entities.Entry{*e1, *e2}
+
+		err = useCase.CreateTransaction(context.Background(), uuid.New(), entries)
+		assert.Nil(t, err)
+	})
+
+	t.Run("Fail with invalid expected version", func(t *testing.T) {
+		useCase := newFakeUseCase(nil)
+
+		accountID1 := uuid.New()
+		accountID2 := uuid.New()
+		e1 := entities.NewEntry(uuid.New(), entities.DebitOperation, accountID1, entities.NewAccountVersion, 123)
+		e2 := entities.NewEntry(uuid.New(), entities.CreditOperation, accountID2, entities.NewAccountVersion, 123)
+		entries := []entities.Entry{*e1, *e2}
+
+		err := useCase.CreateTransaction(context.Background(), uuid.New(), entries)
+
+		assert.Nil(t, err)
+
+		e1 = entities.NewEntry(uuid.New(), entities.DebitOperation, accountID1, 2, 123)
+		e2 = entities.NewEntry(uuid.New(), entities.CreditOperation, accountID2, 4, 123)
+		entries = []entities.Entry{*e1, *e2}
+
+		err = useCase.CreateTransaction(context.Background(), uuid.New(), entries)
+		assert.True(t, errors.Is(err, entities.ErrInvalidVersion))
+	})
+
+	t.Run("When transaction fail, the counter isn't incremented", func(t *testing.T) {
+		useCase := newFakeUseCase(nil)
+
+		accountID1 := uuid.New()
+		accountID2 := uuid.New()
+		e1 := entities.NewEntry(uuid.New(), entities.DebitOperation, accountID1, entities.NewAccountVersion, 123)
+		e2 := entities.NewEntry(uuid.New(), entities.CreditOperation, accountID2, entities.NewAccountVersion, 123)
+		entries := []entities.Entry{*e1, *e2}
+
+		err := useCase.CreateTransaction(context.Background(), uuid.New(), entries)
+
+		assert.Nil(t, err)
+
+		lastVersion := useCase.GetLastVersion()
+
+		e1 = entities.NewEntry(uuid.New(), entities.DebitOperation, accountID1, 2, 123)
+		e2 = entities.NewEntry(uuid.New(), entities.CreditOperation, accountID2, 4, 123)
+		entries = []entities.Entry{*e1, *e2}
+
+		err = useCase.CreateTransaction(context.Background(), uuid.New(), entries)
+		assert.True(t, errors.Is(err, entities.ErrInvalidVersion))
+
+		assert.Equal(t, lastVersion, useCase.GetLastVersion())
+	})
+
+	t.Run("Object version does not change when the transaction fails", func(t *testing.T) {
```

Comment:
The test name should be 'Global version'
```suggestion
	t.Run("Global version does not change when the transaction fails", func(t *testing.T) {
```